### DISCUSSION TITLE
WebButton turns on 2 relays using rules - how?

### DISCUSSION THREAD

**Discussion by CyBaart** [arendst/Tasmota#19348] [Q&A] [answered]
I am making a hot tub controller. I have two web buttons renamed to Pump and Heater. These buttons activate solid state relays on IO25 and IO26. Now, I want to be able tu turn on the pump without the heater BUT when I turn on the heater, the pump MUST turn on also. This is what I tried;

rule1 on power2 1 do power1 1 endon on ds18b20#temperature>=35 do power2 0 endon break

The second 'on' turns the heater off at the required temperature but It doesn't work regarding energizing both relays..

Please advise how this could be done. Thanks, Steve.

**Comment by sfromis** [comment]
Use correct syntax, `power2==1`

**Reply by CyBaart** [reply]
Of course, just like the temperature>=35.  Had I been using an IF statement, I would have done it like that. Thank you.

**Reply by CyBaart** [reply]
Except is still isn't working!

**Reply by sfromis** [reply, marked as answer]
Oops, at the same time as the correct syntax, I also should've mentioned that you need to use the correct trigger name, which would be `Power2#state`

**Comment by pistoletov1974** [comment]
I have the same issue. My Rule is:
`RESULT = {"Rule1":{"State":"ON","Once":"OFF","StopOnError":"OFF","Length":90,"Free":421,"Rules":"Power1#state = ON DO RfSend {\"Data\":\"0xFE8408F0\",\"Bits\":32,\"Protocol\":1,\"Pulse\":254} ENDON"}}`
But if i pres webuton:
`12:29:00.980 RSL: RESULT = {"POWER1":"ON"}
12:29:00.983 RSL: POWER1 = ON
12:29:01.982 RSL: RESULT = {"POWER1":"OFF"}
12:29:01.985 RSL: POWER1 = OFF`

i did not see any info about rule fire in log

**Reply by barbudor** [reply]
Proper rule syntax should be as documented:
```
ON <trigger> DO <command> ENDON
```
from which you own syntax it totally invalid
```
Power1#state = ON DO RfSend {\"Data\":\"0xFE8408F0\",\"Bits\":32,\"Protocol\":1,\"Pulse\":254} ENDON
````

I assume you wanted to write:
```
ON Power1#state==1 DO RfSend ..... ENDON
```
As documented, the power states (or the buttons or the switches) MUST be compared to their numerical equivalent, not the string
ON, OFF, TOGGLE are shortcuts for the command, not what should be used in the trigger

**Reply by pistoletov1974** [reply]
Does information about triggered rules appeared in log?  

`12:45:19.726 CMD: Rule1 Power1#state==1  DO RfSend {"Data":"0xFE8408F0","Bits":32,"Protocol":1,"Pulse":254} ENDON
12:45:19.734 RUL: Stored uncompressed, would compress from 89 to 90 (-0%)
12:45:19.739 RSL: RESULT = {"Rule1":{"State":"ON","Once":"OFF","StopOnError":"OFF","Length":89,"Free":422,"Rules":"Power1#state==1  DO RfSend {\"Data\":\"0xFE8408F0\",\"Bits\":32,\"Protocol\":1,\"Pulse\":254} ENDON"}}`

after press webbutton:

`12:46:13.898 RSL: RESULT = {"POWER1":"ON"}
12:46:13.901 RSL: POWER1 = ON`

**Reply by sfromis** [reply]
Yes, anytime a rule is triggered, you get a message.

And your rule set is still bad, missing the `on` at the start.
`12:45:19.726 CMD: Rule1 Power1#state==1  DO RfSend {"Data":"0xFE8408F0","Bits":32,"Protocol":1,"Pulse":254} ENDON`

**Reply by pistoletov1974** [reply]
Thank you! I resolve my issue!
